build.rs

```diff
@@ -1,46 +1,51 @@
-//! Reads `template/index.ron` and includes the files listed their into the source code.
-//! See `src/new.rs` for how it is used.
-
 extern crate ron;
 
+use std::collections::HashMap;
 use std::env;
-use std::fs::File;
+use std::fs::{read_dir, File};
 use std::io::Write;
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
 
 use ron::de::from_reader;
 
 fn path(env: &str, s: &str) -> PathBuf {
     PathBuf::from(env::var(env).unwrap()).join(s)
 }
 
-fn read_template_index() -> Vec<String> {
-    let path = path("CARGO_MANIFEST_DIR", "template/index.ron");
-    from_reader(File::open(&path).expect("Failed to open `template/index.ron`"))
+fn read_template_index<P: AsRef<Path>>(p: P) -> Vec<String> {
+    let mut path = PathBuf::new();
+    path.push(p);
+    path.push("index.ron");
+    from_reader(File::open(&path).expect("Failed to open index.ron"))
         .expect("Failed to parse template index")
 }
 
 fn main() {
-    let index = read_template_index();
-    let mut source_code = index.iter().fold(
-        "pub fn template_files() -> Vec<(&'static str, &'static str)> {
-    let mut map = Vec::new();\n"
-            .to_owned(),
-        |s, file| {
+    let f = PathBuf::from(path("CARGO_MANIFEST_DIR", "templates"));
+    let mut indices = HashMap::new();
```

Comment:
Can be replaced with functional-style code to make it more readable:

```rust
read_dir(&f)
    .unwrap()
    .map(Result::unwrap)
    .map(|v| (v.file_name().into_string().unwrap(), read_template_index(v.path()))
```

I think you don't even need to build the hash map, you can just iterate over the items.